README.md

```diff
@@ -1,85 +1,95 @@
-auth_status = api.get_authentication_status(auth.id)
-if (auth_status.pending == False and auth_status.granted == True):
+authentication_request.refresh_from_server()
+if not authentication_request.pending and authentication_request.granted:
 	# Success!
 ```
 
-#### Handling Errors
-If any request runs into an error a `ToopherApiError` will be thrown with more details on what went wrong.
+## ToopherIframe Workflow
+### Step 1: Embed a request in an IFRAME
+1. Generate an authentication URL by providing a username.
+2. Display a webpage to your user that embeds this URL within an `<iframe>` element.
+
+```python
+import toopher
+
+# Create an API Object Using Your Credentials
+iframe_api = toopher.ToopherIframe("<your consumer key>", "<your consumer secret>")
+
 
-#### Zero-Storage usage option
-Requesters can choose to integrate the Toopher API in a way does not require storing any per-user data such as Pairing ID and Terminal ID - all of the storage
-is handled by the Toopher API Web Service, allowing your local database to remain unchanged.  If the Toopher API needs more data, it will `raise()` a specific
-error that allows your code to respond appropriately.
+auth_iframe_url = iframe_api.get_authentication_url(username);
+
+# Add an <iframe> element to your HTML:
+# <iframe id="toopher_iframe" src=auth_iframe_url />
+```
+
+### Step 2: Validate the postback data
+
+The simplest way to validate the postback data is to call `is_authentication_granted` to check if the authentication request was granted.
 
 ```python
-try:
-    # optimistically try to authenticate against Toopher API with username and a Terminal Identifier
-    # Terminal Identifer is typically a randomly generated secure browser cookie.  It does not
-    # need to be human-readable
-    auth = api.authenticate_by_user_name(user_name, requester_terminal_id)
-
-    # if you got here, everything is good!  poll the auth request status as described above
-    # there are four distinct errors ToopherAPI can return if it needs more data
-except UserDisabledError:
-    # you have marked this user as disabled in the Toopher API.
-except UserUnknownError:
-    # This user has not yet paired a mobile device with their account.  Pair them
-    # using api.pair() as described above, then re-try authentication
-except TerminalUnknownError:
-    # This user has not assigned a "Friendly Name" to this terminal identifier.
-    # Prompt them to enter a terminal name, then submit that "friendly name" to
-    # the Toopher API:
-    #   api.create_user_terminal(user_name, terminal_name, requester_terminal_id)
-    # Afterwards, re-try authentication
-except PairingDeactivatedError:
-    # this user does not have an active pairing,
-    # typically because they deleted the pairing.  You can prompt
-    # the user to re-pair with a new mobile device.
+# Returns boolean indicating if user should be granted access
+authentication_request_granted = iframe_api.is_authentication_granted(form_data)
```

Comment:
we need to add some code telling them what `form_data` is.  It's just a string from a single POST parameter, right?